packages/@react-aria/landmark/src/useLandmark.ts

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2022 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {AriaLabelingProps} from '@react-types/shared';
+import {getFocusableTreeWalker} from '@react-aria/focus';
+import {HTMLAttributes, MutableRefObject} from 'react';
+import {useLayoutEffect} from '@react-aria/utils';
+
+export interface AriaLandmarkProps  extends AriaLabelingProps {
+  role: 'main' | 'region' | 'search' | 'navigation' | 'form' | 'banner' | 'contentinfo' | 'complementary'
+}
+
+interface LandmarkAria {
+  landmarkProps: HTMLAttributes<HTMLElement>
+}
+
+type Landmark = {
+  ref: MutableRefObject<HTMLElement>,
+  role: string,
+  label?: string,
+  lastFocused?: HTMLElement
+};
+
+class LandmarkManager {
+  private landmarks: Array<Landmark> = [];
+  private static instance: LandmarkManager;
+
+  private constructor() {}
+
+  public static getInstance(): LandmarkManager {
+    if (!LandmarkManager.instance) {
+      LandmarkManager.instance = new LandmarkManager();
+    }
+
+    return LandmarkManager.instance;
+  }
+
+  // Get list of landmarks with a specific role.
+  public getLandmarksByRole(role) {
+    return new Set(this.landmarks.filter(l => l.role === role));
+  }
+
+  public addLandmark(ref: MutableRefObject<HTMLElement>, {role, label}: Omit<Landmark, 'ref' |'lastFocused'>) {
```

Comment:
given that the ref is also something we're putting in the same place, we should just condense the arguments to this function to the single object and only omit lastFocused

Reply:
Good idea, will do